catalogue_of_life/Loader.java

```diff
@@ -0,0 +1,242 @@
+                ex.getCommandLine().usage(commandLine.getErr());
+            }
+            return Optional.empty();
+        }
+    }
+
+    public static Path fetchZip(Path workingDirectory, Options options) throws MalformedURLException {
+        URL url = new URL("https://api.checklistbank.org/dataset/9817/export.zip?format=ColDP");
+
+        Path zipFile = workingDirectory.resolve("coldp.zip");
+        if (!options.skipDownload() && (options.forceDownload() || !Files.exists(zipFile))) {
+            try (InputStream inputStream = url.openStream()) {
+                Files.copy(inputStream, zipFile, StandardCopyOption.REPLACE_EXISTING);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        return zipFile;
+    }
+
+    private static void unzip(Path zipFile, Path workingDirectory) {
+        try (ZipInputStream zipIn = new ZipInputStream(Files.newInputStream(zipFile))) {
+            for (ZipEntry zipEntry; (zipEntry = zipIn.getNextEntry()) != null; ) {
+                Path resolvedPath = workingDirectory.resolve(zipEntry.getName());
+                if (zipEntry.isDirectory()) {
+                    Files.createDirectories(resolvedPath);
+                } else {
+                    Files.createDirectories(resolvedPath.getParent());
+                    Files.copy(zipIn, resolvedPath, StandardCopyOption.REPLACE_EXISTING);
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void prepareData(Path dataDirectory) throws IOException {
+        TsvParser distributionsParser = beginParsingFile(dataDirectory.resolve("Distribution.tsv"));
+        TsvWriter marineRegionsWriter = createTsvWriter(dataDirectory.resolve("MarineDistribution.tsv"));
+        TsvWriter describedRegionsWriter = createTsvWriter(dataDirectory.resolve("DescribedDistribution.tsv"));
+
+        Record record;
+        Map<String, String> recordValues = new HashMap<>();
+        while ((record = distributionsParser.parseNextRecord()) != null) {
+            String gazetteer = record.getString("col:gazetteer");
+            if (gazetteer.equals("text") || gazetteer.equals("iso")) {
+                record.fillFieldMap(recordValues);
+                describedRegionsWriter.writeRow(recordValues);
+            } else if (gazetteer.equals("mrgid") && isValidMarineRegionID(record.getString("col:areaID"))) {
+                record.fillFieldMap(recordValues);
+                marineRegionsWriter.writeRow(recordValues);
+            }
+        }
+
+        marineRegionsWriter.close();
+        describedRegionsWriter.close();
+    }
+
+    private static TsvParser beginParsingFile(Path tsvFile) throws IOException {
+        TsvParserSettings settings = new TsvParserSettings();
+        settings.setHeaderExtractionEnabled(true);
+        settings.setMaxCharsPerColumn(1_000_000);
```

Comment:
1M is arbitrary. Rows get really long here, and the default value of 4096 is just not enough.